src/openrct2/rct1/S4Importer.cpp

```diff
@@ -2624,14 +2623,14 @@ namespace RCT1
             const RCT1::ResearchItem* researchList = GetResearchList(&researchListCount);
             for (size_t i = 0; i < researchListCount; i++)
             {
-                if (researchList[i].flags == 0xFF)
+                if (researchList[i].Flags == 0xFF)
                 {
                     break;
                 }
 
-                if (researchList[i].type == RCT1_RESEARCH_TYPE_RIDE)
+                if (researchList[i].Type == RCT1_RESEARCH_TYPE_RIDE)
                 {
-                    return RCT1::GetRideType(static_cast<RideType>(researchList[i].item), 0);
+                    return RCT1::GetRideType(static_cast<RideType>(researchList[i].Item), 0);
                 }
             }
 
```

Comment:
L2651 should be changed to `if (RCT12PeepType(static_cast<const RCT1::Peep*>(&src)->Type) == RCT12PeepType::Guest)`

Reply:
I used a different solution, one that I thought was less ambiguous.

Reply:
You may as well use the type `RCT12PeepType` for it as well. Could you check rct2's one as i think we will have the same issue.

Reply:
Mercifully, RCT2’s peep has called this field `peep_type`.